crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -161,6 +160,22 @@ module std::vector {
 
     // === Macros ===
 
+    /// Create a vector of length `n` by calling the function `f` on each index.
+    public macro fun tabulate<$T>($n: u64, $f: |u64| -> $T): vector<$T> {
```

Comment:
Another macro to consider is `iota` (or `range)`, which takes `n` and returns a vector from 0 to that index. It would make this a bit redundant, as you could always call `vector::range!(10).map(...)`

Reply:
I like this idea (and like it even more than `tabulate`), so calling in experts like @tnowacki @tzakian and @amnn to comment.